dev-libs/glib/patches/glib2-2.53.4.patchset

```diff
@@ -0,0 +1,246 @@
+From 2107a6fce76fb34082a2fe24aa8814a931cbbb1b Mon Sep 17 00:00:00 2001
+From: Jerome Duval <jerome.duval@gmail.com>
+Date: Sat, 19 Dec 2015 14:51:01 +0000
+Subject: glib_compile_schemas: links against gio, gobject and gmodule.
+
+Otherwise build fails on Haiku.
+
+diff --git a/gio/Makefile.am b/gio/Makefile.am
+index b2db995..c82f05c 100644
+--- a/gio/Makefile.am
++++ b/gio/Makefile.am
+@@ -817,7 +817,11 @@ gio_querymodules_LDADD	 = libgio-2.0.la 		\
+ gconstructor_as_data.h: $(top_srcdir)/glib/gconstructor.h data-to-c.pl
+ 	$(AM_V_GEN) $(srcdir)/data-to-c.pl $(top_srcdir)/glib/gconstructor.h gconstructor_code > $@.tmp && mv $@.tmp $@
+ 
+-glib_compile_schemas_LDADD = $(top_builddir)/glib/libglib-2.0.la
++glib_compile_schemas_LDADD = libgio-2.0.la 		\
++	$(top_builddir)/gobject/libgobject-2.0.la	\
++	$(top_builddir)/gmodule/libgmodule-2.0.la	\
++	$(top_builddir)/glib/libglib-2.0.la		\
++	$(NULL)
+ glib_compile_schemas_SOURCES = \
+ 	gconstructor_as_data.h \
+ 	gvdb/gvdb-format.h		\
+-- 
+2.7.0
+
+
+From 9229a461b8bbfaa0bd8b232ef0013a84392fb1ff Mon Sep 17 00:00:00 2001
+From: Jerome Duval <jerome.duval@gmail.com>
+Date: Sat, 19 Dec 2015 14:59:42 +0000
+Subject: configure.ac: on Haiku
+
+* don't use -pthread
+* use -lbe -lroot as thread related libs.
+
+diff --git a/configure.ac b/configure.ac
+index fbc4400..7386289 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -154,6 +154,9 @@ case "$host" in
+     ;;
+ esac
+ case $host in
++  *-*-haiku*)
++    LIBS="-lnetwork $LIBS"
++    ;;
+   *-*-linux*)
+     glib_os_linux=yes
+     ;;
+@@ -1922,8 +1925,8 @@ AS_IF([ test x"$have_threads" = xposix], [
+   # One of them does for most gcc versions and some other platforms/compilers
+   # too and could be considered as the canonical way to go. 
+   case $host in
+-    *-*-cygwin*|*-*-darwin*)
+-       # skip cygwin and darwin -pthread or -pthreads test
++    *-*-cygwin*|*-*-darwin*|*-*-haiku*)
++       # skip cygwin, darwin and haiku -pthread or -pthreads test
+        ;;
+     *-solaris*)
+       # These compiler/linker flags work with both Sun Studio and gcc
+@@ -2088,6 +2091,15 @@ if test "x$G_THREAD_LIBS" = xerror; then
+         AC_MSG_ERROR($LIBS_NOT_FOUND_1$have_threads$LIBS_NOT_FOUND_2)
+ fi
+ 
++case $host in
++  *-*-haiku*)
++    G_THREAD_LIBS="-lbe -lroot"
++    G_THREAD_LIBS_EXTRA="-L\$(top_builddir_full)/.libs"
++    ;;
++  *)
++    ;;
++esac
++
+ AC_MSG_CHECKING(thread related libraries)
+ AC_MSG_RESULT($G_THREAD_LIBS)
+ 
+-- 
+2.7.0
+
+
+From 2ae7f7f265f807896f61a19ad715d33a6ac0e6b2 Mon Sep 17 00:00:00 2001
```

Comment:
This one could use some looking into, I tried to re-implement the patch from previous version, but things have changed since then ...